app/src/main/java/com/cornellappdev/scoop/ui/components/search/DisplaySearchesPage.kt

```diff
@@ -28,20 +28,22 @@ import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
 import com.cornellappdev.scoop.R
 import com.cornellappdev.scoop.data.models.Ride
-import com.cornellappdev.scoop.data.models.Search
 import com.cornellappdev.scoop.ui.components.general.FilterRow
 import com.cornellappdev.scoop.ui.components.general.RideCard
+import com.cornellappdev.scoop.ui.viewmodel.SearchScreenViewModel
 
 /**
  * This page allows users to look at their search, filter and view their results.
  *
  * Users also have the ability to edit their search.
  *
- * @param searchState State that represents the current search the user inputted
+ * @param searchScreenViewModel ViewModel that represents the current search the user inputted
  */
 @SuppressLint("UnusedMaterialScaffoldPaddingParameter")
 @Composable
-fun DisplaySearchesPage(searchState: MutableState<Search>) {
+fun DisplaySearchesPage(
+    searchScreenViewModel: SearchScreenViewModel
```

Comment:
Sorry small comment, viewmodels should not be created like this but should be referenced using viewModel(), see[ this link](https://developer.android.com/jetpack/compose/libraries#viewmodel) for an example.

hiltViewModel() returns an existing ViewModel or creates a new one in the given scope. The ViewModel is retained as long as the scope is alive. For example, if the composable is used in an activity, viewModel() returns the same instance until the activity is finished or the process is killed.

This is something Chris brought to my attention with Justin's code a while back. You will actually notice you do it on your PostScreen. It just makes it easier so you don't have to pass in the viewModel into everything for it to work

Reply:
What do you mean? In the subpages, should I be referring to the view model instead of passing it in?

Reply:
If you look at your PostScreenViewModel,
`@HiltViewModel
class PostScreenViewModel @Inject constructor(private val rideRepository: RideRepository) :
    ViewModel() {`
You add a @HiltViewModel onto the viewModel
This way, you can get the postScreenViewModel here just by calling the function. This way you don't have to pass the viewModel into the parameters of your code making it cleaner and easier to write
`@Composable
fun PostScreen(
    onPostNewTrip: (RideRequestBody) -> Unit,
    postScreenViewModel: PostScreenViewModel = hiltViewModel()
) {`